types/react/index.d.ts

```diff
@@ -298,9 +303,17 @@ declare namespace React {
         // always pass children as variadic arguments to `createElement`.
         // In the future, if we can define its call signature conditionally
         // on the existence of `children` in `P`, then we should remove this.
-        props: Readonly<{ children?: ReactNode }> & Readonly<P>;
-        state: Readonly<S>;
+        readonly props: Readonly<{ children?: ReactNode }> & Readonly<P>;
+        readonly state: null | Readonly<S>;
```

Comment:
Should this instead be:
```ts
readonly state?: ReadOnly<S>;
```

React itself does not instantiate state on its own (not even to `null`), and this more accurately depicts what actually happens (a consumer does not always have to instantiate `this.state`)

Reply:
you're wrong, state is `null` by default ✌️

![image](https://user-images.githubusercontent.com/1223799/41856341-1d6d3534-7895-11e8-9470-b55a9608d872.png)

Reply:
Ooh! TIL!

Reply:
Looks like instantiating state with null creates an issue when one sets state in a constructor.